aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/DownloadCallable.java

```diff
@@ -197,7 +196,10 @@ private void downloadInParallel(int partCount) throws Exception {
      * Merges all the individual part Files into dstFile
      */
     private void combineFiles() throws Exception {
-        truncateDestinationFileIfNecessary();
+        if (!ServiceUtils.createParentDirectoryIfNecessary(dstfile)) {
```

Comment:
The future file get also creates the parent directory if necessary I believe. So which ever one starts first should create the parent directory and hopefully resolve the issue.